src/platform/common/stm32/adc_impl.c

```diff
@@ -0,0 +1,104 @@
+/*
+ * This file is part of Betaflight.
+ *
+ * Betaflight is free software. You can redistribute this software
+ * and/or modify this software under the terms of the GNU General
+ * Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later
+ * version.
+ *
+ * Betaflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdbool.h>
+
+#include "drivers/adc.h"
+#include "platform/adc_impl.h"
+
+// Verify a pin designated by tag has connection to an ADC instance designated by device
+bool adcVerifyPin(ioTag_t tag, adcDevice_e device)
+{
+    if (!tag) {
+        return false;
+    }
+
+    for (int map = 0 ; map < ADC_TAG_MAP_COUNT ; map++) {
```

Comment:
maybe:
```suggestion
    for (int map = 0 ; map < ARRAYLEN(adcTagMap); map++) {
```
Or you can iterate entries directly:
```suggestion
    for (const adcTagMap_t m = adcTagMap; m < ARRAYEND(adcTagMap); m++) {
```